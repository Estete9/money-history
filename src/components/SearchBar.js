import { useSelector, useDispatch } from 'react-redux';
import { toggleSearchBar, updateSearchQuery } from '../redux/header/headerSlice';
import closeBtn from '../assets/close.svg';
import styles from '../styles/searchBar.module.css';

function SearchBar() {
  const { isSearchBarOpen, searchQuery } = useSelector((store) => store.header);
  const dispatch = useDispatch();

  const handleCloseBtn = () => {
    dispatch(toggleSearchBar());
  };

  const handleSearchQuery = (event) => {
    dispatch(updateSearchQuery(event.target.value));
  };

  // useEffect((queryValue) => {
  //   dispatch(updateSearchQuery(queryValue));
  // });

  return (
    isSearchBarOpen && (
      <div className={`${styles.searchBarWrapper} ${isSearchBarOpen ? styles.show : ''}`}>
        <input
          className={styles.searchBar}
          placeholder="Select one of the 8 available currencies"
          type="search"
          value={searchQuery}
          onChange={(e) => handleSearchQuery(e)}
        />
        <button type="button" className={styles.closeBtn} onClick={handleCloseBtn}>
          <img className={styles.closeBtnImg} src={closeBtn} alt="closeBtn icon" />
        </button>
      </div>
    )
  );
}

export default SearchBar;
