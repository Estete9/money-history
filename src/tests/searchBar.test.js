import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SearchBar from '../components/SearchBar';

describe('Integration test for searchBar component', () => {
  test('SearchBar is displayed correctly', () => {
    const mockStore = configureStore([]);
    const store = mockStore({
      header: {
        isSearchBarOpen: true,
        searchQuery: '',
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <SearchBar />
      </Provider>,
    );
    const searchBarTree = component.toJSON();
    expect(searchBarTree).toMatchSnapshot();
  });
});
