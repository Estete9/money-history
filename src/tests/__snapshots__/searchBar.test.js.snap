// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration test for searchBar component SearchBar is displayed correctly 1`] = `
<div
  className="searchBarWrapper show"
>
  <input
    className="searchBar"
    onChange={[Function]}
    placeholder="Search currency code (E.g. USD)"
    type="search"
    value=""
  />
  <button
    className="closeBtn"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="closeBtn icon"
      className="closeBtnImg"
      src="svg-mock"
    />
  </button>
</div>
`;
